src/sentry/incidents/logic.py

```diff
@@ -936,22 +936,37 @@ def update_alert_rule(
                 raise ResourceDoesNotExist(
                     "Your organization does not have access to this feature."
                 )
-
             if updated_fields.get("detection_type") == AlertRuleDetectionType.DYNAMIC and (
                 alert_rule.detection_type != AlertRuleDetectionType.DYNAMIC or query or aggregate
             ):
                 for k, v in updated_fields.items():
                     setattr(alert_rule, k, v)
 
+                for k, v in updated_query_fields.items():
```

Comment:
any snuba query updates were previously not getting passed to the Seer code until this was added

Reply:
That check is saying "if the rule type is changing to become a dynamic type OR it's already dynamic and the update changes either the query or the aggregate..." then we need to refetch the historical data and send it to Seer because either it wasn't a dynamic type before (so Seer wouldn't have that data) or the snuba query has changed so the historical data has changed